### ISSUE TITLE
ruby-head does not install anymore due to missing tcltk libraries

### ISSUE THREAD

**Issue by eregon** [travis-ci/travis-ci#5166]
```
Libraries missing for ruby-head: libtcl8.5.so.0,libtk8.5.so.0.
Refer to your system manual for installing libraries
Mounting remote ruby failed with status 10, stopping installation.
```

See https://travis-ci.org/ruby/rubyspec/jobs/93116606 for example.

**Comment by shinnn** [comment]
As a workaround, I tried to install `tcltk` by:

```yaml
addons:
  apt:
    packages: tcltk
rvm: ruby-head
```

But `tcltk` hasn't been approved yet.

```
Installing APT Packages (BETA)
Disallowing packages: tcltk
If you require these packages, please review the package approval process at: https://github.com/travis-ci/apt-package-whitelist#package-approval-process
```

**Comment by BanzaiMan** [comment]
Just FYI, `tcl` and `tk` are separate packages on Ubuntu. http://packages.ubuntu.com/precise/tcl http://packages.ubuntu.com/precise/tk

**Comment by BanzaiMan** [comment]
This is quite curious, since we have not changed the way we package ruby-head. I wonder what caused the problem. :-/

**Comment by shinnn** [comment]
Ugh, really shameful. :sweat_smile:

**Comment by eregon** [comment]
Maybe some bad detection in Ruby's configure?

**Comment by BanzaiMan** [comment]
@eregon Could be. You can look at how we build it with https://github.com/travis-ci/travis-rubies/blob/build/build.sh

**Comment by eregon** [comment]
@BanzaiMan The failure only happens at verification, it seems ruby-head is built successfully by that script.
Could you verify if these packages are installed in the CI image? And what version?
Or look at what changed recently in VM image generation?

I also tried to install the packages dynamically in https://travis-ci.org/ruby/rubyspec/jobs/93347137 but they were not allowed.

**Comment by BanzaiMan** [comment]
@eregon Right. There is a process for adding packages to the list: f you require these packages, please review the package approval process at: https://github.com/travis-ci/apt-package-whitelist#package-approval-process

The script is designed to build binaries statically. I think it used to work, but perhaps it is not longer the case. I reckon this is the same issue discussed in https://github.com/travis-ci/travis-ci/issues/4984.

**Comment by eregon** [comment]
Downloading the binary indeed gives:
```
$ ldd lib/ruby/2.3.0/x86_64-linux/tcltklib.so
	linux-vdso.so.1 (0x00007fffec5d1000)
	libruby.so.2.3 => not found
	libtcl8.5.so.0 => not found
	libtk8.5.so.0 => not found
	libc.so.6 => /lib64/libc.so.6 (0x00007f9f347df000)
	/lib64/ld-linux-x86-64.so.2 (0x000055bc9109c000)
```

**Comment by eregon** [comment]
That said, I'll let you handle guys handle this, since it seems at the rvm layer.

**Comment by eregon** [comment]
@BanzaiMan It looks like it works now, thanks!
https://travis-ci.org/ruby/rubyspec/jobs/93664719 passed for instance.

**Comment by BanzaiMan** [comment]
Great to hear that my fix was effective.

It turns out that we actually _did_ change the `ruby-head` packaging process; it was moved to the new Precise image (see [this blog post](https://blog.travis-ci.com/2015-11-27-moving-to-a-more-elastic-future)), which has Tcl/Tk pre-installed, and `ruby-head` was built with that feature enabled.

It is not clear to me why this discrepancy exists (or it is even possible to know; Tcl/Tk might be an indirect dependency of software we added to the GCE Precise image).

Given that the lack of Tcl/Tk support has never been an issue before, I tend to think this is still not a problem. We are now explicitly disabling Tcl/Tk support (at least until such time that **all** of our infrastructures—GCE, containers—have such support) on all new Ruby builds.
